flask_rebar/swagger_generation/swagger_generator.py

```diff
@@ -668,18 +655,303 @@ def _get_definitions(self, paths):
+                            in_=sw.query
+                        )
+                    )
+
+                if path_definition.request_body_schema:
+                    schema = path_definition.request_body_schema
+
+                    parameters.append(
+                        oa.Parameter(
+                            name=schema.__class__.__name__,
+                            in_=sw.body,
+                            required=True,
+                            schema=self._get_schema(schema),
+                        )
+                    )
+
+                if path_definition.headers_schema is USE_DEFAULT and default_headers_schema:
+                    parameters.extend(
+                        self._convert_jsonschema_to_list_of_parameters(
+                            self._headers_converter(default_headers_schema),
+                            in_=sw.header
+                        )
+                    )
+                elif path_definition.headers_schema is not USE_DEFAULT and path_definition.headers_schema is not None:
+                    parameters.extend(
+                        self._convert_jsonschema_to_list_of_parameters(
+                            self._headers_converter(path_definition.headers_schema),
+                            in_=sw.header
+                        )
+                    )
+
+                security = []
+                if path_definition.authenticator is not None and path_definition.authenticator is not USE_DEFAULT:
+                    security = [self._get_security(path_definition.authenticator)]
+
+                operation = oa.Operation(
+                    responses=responses,
+                    operation_id=path_definition.endpoint or get_swagger_title(path_definition.func),
+                    description=path_definition.func.__doc__,
+                    tags=path_definition.tags,
+                    parameters=parameters,
+                    security=security,
+                )
+
+                setattr(path_item, method.lower(), operation)
+
+        return paths
+
+    @staticmethod
+    def _convert_jsonschema_to_list_of_parameters(obj, in_='query'):
+        """
+        Swagger is only _based_ on JSONSchema. Query string and header parameters
+        are represented as list, not as an object. This converts a JSONSchema
+        object (as return by the converters) to a list of parameters suitable for
+        swagger.
+
+        :param dict obj:
+        :param str in_: 'query' or 'header'
+        :rtype: list[dict]
+        """
+        parameters = []
+
+        assert obj['type'] == 'object'
+
+        required = obj.get('required', [])
+
+        for name, prop in sorted(obj['properties'].items(), key=lambda i: i[0]):
+            schema = oa.Schema()
+            for k, v in prop.items():
+                python_key = _get_python_key_from_swagger_word(k)
+                setattr(schema, python_key, v)
+
+            parameter = oa.Parameter(
```

Comment:
I don't think you can instantiate marshmallow schemas like this - `Schema` may not be the object you're looking for.

You might want to opt for regular ol' classes here. Or maybe get fancy with a new type.